hw/ip/pattgen/data/pattgen.hjson

```diff
@@ -0,0 +1,148 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+{ name: "pattgen"
+  clock_primary: "clk_fixed"
+  bus_device: "tlul"
+  bus_host: "none"
+  estimated_gate: "TBD"
+  // OUTPUT pins
+  available_output_list: [
+    { name: "pda1_tx", desc: "Serial output data bit for pattern generation on channel 1" }
+    { name: "pcl1_tx", desc: "Bit-banging output clock for pattern generation on channel 1" }
+    { name: "pda2_tx", desc: "Serial output data bit for pattern generation on channel 2" }
+    { name: "pcl2_tx", desc: "Bit-banging output clock for pattern generation on channel 2" }
+  ]
+  // INTERRUPT pins
+  interrupt_list: [
+    { name: "pattern_done"
+      desc: "raise if pattern generation is complete"
+    }
+  ]
+
+  // REGISTER definition
+  regwidth: "32"
+  registers: [
+    // CTRL register
+    { name: "CTRL"
+      desc: "PATTGEN control register (Functions TBD)"
+      swaccess: "rw"
+      hwaccess: "hro"
+      fields: [
+        { bits: "0"
+          resval: "0"
+          name: "ENABLE"
+          desc: '''
+                Enable pattern generator functionality
+                '''
+        }
+      ]
+    }
+    { name: "START"
+      desc: "PATTGEN start pattern generation"
+      swaccess: "wo"
+      hwaccess: "hro"
+      hwqe: "true"
+      fields: [
+        { bits: "0"
+          resval: "0"
+          name: "START0"
+          desc: "Start generating pattern on channel 1"
+        }
+        { bits: "1"
+          resval: "0"
+          name: "START1"
+          desc: "Start generating pattern on channel 1"
+        }
+      ]
+    }
+    { name: "PREDIV1"
+      desc: "PATTGEN pre-divider of I/O clock for chanel 1 (base is 2)"
+      swaccess: "rw"
+      hwaccess: "hro"
+      hwqe: "true"
+      fields: [
+        { bits: "31:0"
+          resval: "0"
+          name: "RATIO"
+          desc: "Dividend ratio for channel 1 (base is 2)"
+        }
+      ]
+    }
+    { name: "PREDIV2"
+      desc: "PATTGEN pre-divider of I/O clock for chanel 2 (base is 2)"
+      swaccess: "rw"
+      hwaccess: "hro"
+      hwqe: "true"
+      fields: [
+        { bits: "31:0"
+          resval: "0"
+          name: "RATIO"
+          desc: "Dividend ratio for channel 2 (base is 2)"
+        }
+      ]
+    }
+    { name: "PTTLEN"
```

Comment:
I'd rather either always shorten as `PATT` (rather than `PTT`) or
just spell out as `pattern_length`.

Reply:
Rename to PATT_LEN